src/Controls/src/Core/Border/Border.cs

```diff
@@ -64,6 +67,12 @@ void NotifyStrokeShapeChanges()
 
 			if (strokeShape is VisualElement visualElement)
 			{
+				if (_parentRef is null)
+					_parentRef = new WeakReference<Element>(this);
+
+				_parentRef.TryGetTarget(out Element? parent);
+				visualElement.Parent = parent;
```

Comment:
So let's say the `StrokeShape` is being set from a `Style` defined at `Application.Currrent.Resource` level, and let's say this style is being applied to many borders on many pages, potentially many times on each page.

Each time that reference is being used we are overriding the previous parent. So the last created `Border` will set the parent on a `Shape` which is used somewhere else, therefore messing up with the eventual `Binding`.

I've already explained this here: https://github.com/dotnet/maui/issues/18925#issuecomment-1824058832

Until/unless we create a deep copy of the `StrokeShape` while applying it through a `Style` this issue will not be solved.

Reply:
You're right. We have to analyze and decide the path to take. I have created a comment on [X](https://github.com/dotnet/maui/issues/21747#issuecomment-2072303550) to continue the conversation.